<!--igvc_run.launch-->
<launch>
    <master auto="start"/>
<!-- =============   TRANSFORMS  ======================= -->
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_velodyne"
        args="0.1 0.1 1.14 0 0 0 base_link velodyne 100" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_to_gps_frame"
        args="-0.52 0.4 0.92 0 0 0 base_link gps_frame 100" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_to_imu_frame"
        args="0 0 0 0 0 0 base_link imu_frame 100" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_to_left_camera_frame"
        args="0.27 0.06 1.08 0 0.75 0 base_link left_camera_frame 100" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_to_right_camera_frame"
        args="0.27 -0.06 1.08 0 0.75 0 base_link right_camera_frame 100" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_to_left_wheel_frame"
        args="0.27 0.28 -0.47 0 0 0 base_link left_wheel_frame 100" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_to_right_wheel_frame"
        args="0.27 -0.28 -0.47 0 0 0 base_link right_wheel_frame 100" />

  <node pkg="tf" type="static_transform_publisher" name="base_footprint_to_base_link"
        args="0 0 -0.44 0 0 0 base_footprint base_link 100" />

  <node pkg="tf" type="static_transform_publisher" name="map_to_odom"
        args="0 0 0 0 0 0 map odom 100" />

  <node pkg="tf" type="static_transform_publisher" name="odom_to_utm"
        args="-319600 -4726900 0 0 0 0 odom utm 100" />
<!--
      x: 320156.994786 --- 319600
      y: 4727417.997   --- 4726900
-->
<!-- =============   Nodes  ======================= 

    <include file="$(find igvc_run)/launch/navsat_transform.launch">
    </include>
-->
<node name="gps_conv" pkg="gps_common" type="utm_odometry_node">
  <remap from="odom" to="vo"/>
  <remap from="fix" to="/gps/fix" />
  <param name="rot_covariance" value="99999" />
  <param name="frame_id" value="base_footprint" />
</node>

  <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    <param name="output_frame" value="odom"/>
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="false"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="true"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="false"/>

    <remap from="odom" to="wheels/odom"/>
    <!--remap from="robot_pose_ekf/odom_combined" to="odometry/filtered"/-->
  </node>
 

  <include file="$(find igvc_roboteq)/launch/roboteq_driver.launch">
    </include>

  <include file="$(find velodyne_bringup)/launch/move_base.launch">
    </include>

<node name="odom_ekf" pkg="igvc_run" type="odom_ekf.py" output="screen"/>

<node name="novatel_CNS5000" pkg="gps_imu" type="usma_novatel_driver.py" output="screen"/>

</launch>

